entity-service-api/src/main/proto/org/hypertrace/entity/query/service/v1/entity_query_request.proto

```diff
@@ -52,6 +62,19 @@ message SetAttribute {
   LiteralConstant value = 2;
 }
 
+message AttributeUpdateOperation {
+  ColumnIdentifier attribute = 1;
+  AttributeUpdateOperator operator = 2;
+  LiteralConstant value = 3;
+
+  enum AttributeUpdateOperator {
+    ATTRIBUTE_UPDATE_OPERATION_UNSPECIFIED = 0;
+    ATTRIBUTE_UPDATE_OPERATOR_SET = 1;
+    ATTRIBUTE_UPDATE_OPERATOR_ADD_TO_LIST_IF_ABSENT = 2;
+    ATTRIBUTE_UPDATE_OPERATOR_REMOVE_ALL_FROM_LIST = 3;
```

Comment:
What would be the `LiteralConstant value` in this case? Is it something like UNSET?

Should we instead do the following?

`SET` -> sets an attribute value. Can it also set an entire array attribute?
`UNSET` -> unsets an attribute value. Basically, removes the attribute. This wouldn't need literal constant being set.
`ADD_TO_LIST` -> adds multiple items to list
`REMOVE_FROM_LIST` -> removes multiple items from list

Reply:
Definition of operators.
`SET` -> sets an attribute value. The value can either be a single value or a complete array.
`UNSET` -> not added here, but document store has support (in this PR: https://github.com/hypertrace/document-store/pull/145/files) and so we can add it if required. Note: For this, the literal value is optional and will be ignored.
`ADD_TO_LIST_IF_ABSENT` -> can either add single or multiple items to a list (add only those which are not already present to be consistent with the existing "bulkUpdateEntityArrayAttribute" API)
`ADD_TO_LIST` -> can directly add either a single or multiple items to a list -- We can add this support if required since document-store supports it (in this PR: https://github.com/hypertrace/document-store/pull/145/files)
`REMOVE_FROM_LIST` -> can either remove all the occurrences of single or multiple items from a list

Reply:
Are you proposing we convert each operator to a one of message?

Reply:
> Are you proposing we convert each operator to a one of message?

We can. Could be an overkill. What do you think?

Reply:
> Could be an overkill.

Agree. Keeping it as an operator makes the mapping easier. Whether to set the `LiteralConstant` etc. would be implied by the operator itself.

> rename REMOVE_ALL_FROM_LIST to REMOVE_FROM_LIST, and may be have an UNSET option too

Sure, will add.